### PR TITLE
Submit progress info

### PR DESCRIPTION
**Pull request by zinkko**
progress information from submitting exercises

### CODE REVIEW

File: src/main/java/fi/helsinki/cs/tmc/core/TmcCore.java [hunk 1]
```diff
@@ -188,22 +188,33 @@ public Path getExerciseChecksumCacheLocation() {
...
+     * @throws TmcCoreException if there was no course in the given path, no exercise in the given
+     * path, or not logged in
+     */
+    public ListenableFuture<SubmissionResult> submit(Path path, ProgressObserver observer) throws TmcCoreException {
         UrlCommunicator communicator = new UrlCommunicator(settings);
         TmcApi tmcApi = new TmcApi(communicator, settings);
 
+        ExerciseSubmitter exerciseSubmitter = new ExerciseSubmitter(
+                new ProjectRootFinder(tmcApi),
+                new StudentFileAwareZipper(new EverythingIsStudentFileStudentFilePolicy()),
```
Comment (jamo) [inline src/main/java/fi/helsinki/cs/tmc/core/TmcCore.java]: Not from this PR, but we probably don't want to submit everything...

Comment (nygrenh) [inline src/main/java/fi/helsinki/cs/tmc/core/TmcCore.java]: We also probably don't want to use those internal classes from langs.

File: src/main/java/fi/helsinki/cs/tmc/core/communication/ExerciseSubmitter.java [hunk 2]
```diff
@@ -136,6 +151,7 @@ public String submitPasteWithComment(String currentPath, String comment)
     }
 
     /**
+<<<<<<< HEAD
```
Comment (nygrenh) [inline src/main/java/fi/helsinki/cs/tmc/core/communication/ExerciseSubmitter.java]: We don't want this

File: src/main/java/fi/helsinki/cs/tmc/core/communication/ExerciseSubmitter.java [hunk 3]
```diff
@@ -153,13 +169,16 @@ public String submitWithCodeReviewRequest(Path currentPath, String message)
     /**
      * Search exercise and throw exception if exercise is expired or not
      * returnable.
+=======
+     * Search exercise and throw exception if exercise is expired or not returnable.
+>>>>>>> submission progress info
```
Comment (nygrenh) [inline src/main/java/fi/helsinki/cs/tmc/core/communication/ExerciseSubmitter.java]: .

File: src/test/java/fi/helsinki/cs/tmc/core/commands/SubmitTest.java [hunk 4]
```diff
@@ -29,30 +29,38 @@
 import com.google.common.util.concurrent.FutureCallback;
 import com.google.common.util.concurrent.Futures;
 import com.google.common.util.concurrent.ListenableFuture;
+import fi.helsinki.cs.tmc.core.domain.ProgressObserver;
```
Comment (nygrenh) [inline src/test/java/fi/helsinki/cs/tmc/core/commands/SubmitTest.java]: Wrong import order

**Pull request merged**